### PR TITLE
Implement CLI and API docker stubs

### PR DESCRIPTION
**Pull request by abalakh**
```python
py.test tests/foreman/api/test_docker.py tests/foreman/cli/test_docker.py -k 'test_positive_pull_image or test_positive_upload_image or test_positive_create_with_external_registry'
================================ test session starts =================================
platform linux2 -- Python 2.7.10, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
collected 96 items 

tests/foreman/api/test_docker.py .
tests/foreman/cli/test_docker.py ...

 92 tests deselected by '-ktest_positive_pull_image or test_positive_upload_image or test_positive_create_with_external_registry' 
===================== 4 passed, 92 deselected in 229.09 seconds ======================
```

### CODE REVIEW

File: tests/foreman/api/test_docker.py [hunk 1]
```diff
@@ -1346,10 +1343,25 @@ def test_positive_create_with_external_registry(self):
...
+        try:
+            compute_resource = entities.DockerComputeResource(
+                organization=[self.org],
+                url=settings.docker.get_unix_socket_url(),
+            ).create()
+            container = entities.DockerRegistryContainer(
+                compute_resource=compute_resource,
+                organization=[self.org],
+                registry=registry,
+                repository_name='rhel',
```
Comment (oshtaier) [inline tests/foreman/api/test_docker.py]: `repository_name=repo_name`?

File: tests/foreman/api/test_docker.py [hunk 2]
```diff
@@ -1346,10 +1343,25 @@ def test_positive_create_with_external_registry(self):
         @Assert: The docker container is created and the image is pulled from
         the external registry
 
-        @caseautomation: notautomated
-
         @CaseLevel: Integration
         """
+        repo_name = 'rhel'
```
Comment (oshtaier) [inline tests/foreman/api/test_docker.py]: Just 'rhel'? Or we can use any specific constant?

**Comment by oshtaier** [comment]
Cool PR. :+1: 
ACK

File: tests/foreman/api/test_docker.py (hunk 1 above)
Comment (abalakh) [inline tests/foreman/api/test_docker.py]: yeah, just mentioned and wanted to quickly fix it while no one noticed... wasted :smile:

**Comment by abalakh** [comment]
Depends on SatelliteQE/nailgun#301

File: tests/foreman/api/test_docker.py (hunk 2 above)
Comment (abalakh) [inline tests/foreman/api/test_docker.py]: any repo from our external registry. as it's red hat one, no busybox is in there for obvious reasons, and frankly i'm not aware which repo is pretty stable to use, but i believe that `rhel` should be there for sure, so we can safely use it :)

**Comment by abalakh** [comment]
Oh, almost forgot - closes #2705. Finally :)

**Pull request merged**